proto/encore/parser/schema/v1/schema.proto

```diff
@@ -4,88 +4,141 @@ option go_package = "encr.dev/proto/encore/parser/schema/v1";
 
 package encore.parser.schema.v1;
 
+// Type represents the base of our schema on which everything else is built on-top of. It has to be one, and only one,
+// thing from our list of meta types.
+//
+// A type may be concrete or abstract, however to determine if a type is abstract you need to recursive through the
+// structures looking for any uses of the TypeParameterPtr type
 message Type {
   oneof typ {
-    Named named = 1;
-    Struct struct = 2;
-    Map map = 3;
-    List list = 4;
-    Builtin builtin = 5;
+    /* Concrete / non-parameterized Types */
+    Named            named          = 1; // A "named" type (https://tip.golang.org/ref/spec#Types)
+    Struct           struct         = 2; // The type is a struct definition
+    Map              map            = 3; // The type is a map
+    List             list           = 4; // The type is a slice
+    Builtin          builtin        = 5; // The type is one of the base built in types within Go
+
+    /* Abstract Types */
+    TypeParameterRef type_parameter = 6; // This is placeholder for a unknown type within the declaration block
```

Comment:
This is new and represents there a placeholder type has been used inside a struct, i.e. the `T` in;

```go
struct {
   field1 T
}
```